services/optimiser/optimiser.py

```diff
@@ -204,6 +211,18 @@ def save_result(self, result) -> None:
                                 'shift_end': shift.endTime
                             })
 
+                            # Fetch FCM tokens for the user and send notification
+                            try:
+                                fcm_token_list = fcm_token_repo.get_fcm_token(user_id=user.id)
+                                if fcm_token_list:
+                                    title = "New shift assignment"
+                                    body = "You have been assigned to a new shift"
+
+                                    # Send notification
+                                    notification_service.send_notification(fcm_token_list=fcm_token_list, title=title, body=body)
```

Comment:
I would call the notification repository function that accepts a user ID, lets you skip manually fetching the fcm tokens